docs/contributing/Regular-Expression-Regex.html

```diff
@@ -0,0 +1,1140 @@
+---
+layout: page
+title: Regular Expression - Regex
+---
+
+<p>The purpose of this page is to provide detailed information for using Regex while editing a masterlist, as well as searching specific terms within it. Regular expressions can formally be described as follows:</p>
+
+<blockquote cite="https://en.wikipedia.org/wiki/Regular_expression">
+A regular expression (shortened as regex or regexp) is a sequence of characters that specifies a search pattern in text.
+</blockquote>
+
+<h1>General</h1>
+<ul>
+  <li>Overview of Regular Expressions: <a href="https://www.rexegg.com/regex-quickstart.html">Quick-Start: Regex Cheat Sheet</a></li>
+  <li>Quickly test your Regex: <a href="https://regex101.com/">Regular Expressions 101</a></li>
+</ul>
+
+<h2>Advantages of Regex</h2>
+Let's take a look at <a href="https://www.nexusmods.com/skyrimspecialedition/mods/41713/">DragonPriest Retexture</a> for a first example:
+
+<p>The DragonPriest Retexture mod includes the two different plugins <code>AcolyteMasks.esl</code> and <code>AcolyteMasks.esp</code>. If we want to add a <code>url</code> for each plugin, we could write the masterlist entry like this:<p>
+
+<pre><code>
+  - name: 'AcolyteMasks.esl'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
+        name: 'DragonPriest Retexture'
+
+  - name: 'AcolyteMasks.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
+        name: 'DragonPriest Retexture'
+</code></pre>
```

Comment:
For prettier code blocks, you can use Jekyll's `{% highlight %}` tag. To actually render the highlighting, though, you would need to add the relevant classes to the site-wide CSS.

For this instance, the code block would look like this:
```suggestion
<pre>{% highlight yaml %}
  - name: 'AcolyteMasks.esl'
    url:
      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
        name: 'DragonPriest Retexture'

  - name: 'AcolyteMasks.esp'
    url:
      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
        name: 'DragonPriest Retexture'
{% endhighlight %}</pre>
```
(don't worry - Jekyll adds the `<code>` tag back)

The classes I've found:
* `pi` - Characters with special meaning such as `:` , `>`, `|`, and `-` outside of a string
* `s1` - Single-Quotes (`'`)
* `s2` - Double-Quotes (`"`)
* `na` - Keys
* `s` - String or Number values
* `nv` - Appears to be whitespace within a quoted string
* `no` - `null` and similar values

Reply:
yaml highlighting would be great indeed, and our [Masterlist Editing](https://loot.github.io/docs/contributing/Masterlist-Editing.html) file could profit from that as well. However, I'm not sure how to go about this, specifically how I would integrate Jekyll support in order to use these `{% highlight yaml %}{% endhighlight %}` tags.

There's the `\css\style.css` file .. but I don't have experience with that.

Reply:
Jekyll is the site builder that you're already using (it's built into GitHub Pages)—that's what the `layout: page` stuff at the top of the document is all about. The tag, for our purposes, is plug-and-play. As for the CSS, I won't be around for about a week, so I unfortunately can't be the one to write it. You could probably get someone—possibly on the Discord server—to write it though!

Reply:
Personally, I find that the easiest way to run Jekyll on Windows is using Docker, for which I use the following command:

```
docker run --rm -v ${PWD}:/srv/jekyll -v jekyll_bundle:/usr/local/bundle -p 4000:4000 -it jekyll/jekyll:3.8.6 jekyll serve --no-watch
```

Reply:
I've heard of Docker before, but haven't used that one before either. I'm open to update this file at a later date to implement the `yaml` highlighting, however I would need help for that as I currently don't understand all the implications on working with CSS/Jekyll/Docker. If you've got the time @BellCubeDev and are interested, please ping me once you are around. 😄